AntennaTracker/system.cpp

```diff
@@ -191,6 +192,11 @@ void Tracker::set_mode(Mode &newmode, const ModeReason reason)
     gcs().send_message(MSG_HEARTBEAT);
 
     nav_status.bearing = ahrs.yaw_sensor * 0.01f;
+
+    // reset the auto mode state
+    if (&mode_auto == &newmode) {
+        mode_auto.switch_state(ModeAuto::State::IDLE);
+    }
```

Comment:
We don't have it in tracker. On other vehicles we typically have a mode enter function where you can do mode specific initialisations.

Reply:
@IamPete1 what's the best way to ensure that the state is reset upon entering the mode then? I was looking for some enter method to be honest and this part felt bad to write.

My rationale for the whole thing is so that if you do e.g. `AUTO` ⇒ `MANUAL` ⇒ `AUTO` transition, you get a state change announcement. But maybe I'm overthinking and this whole thing can be deleted, after all 99% of the time you want to sit in `AUTO` and never change out of it.

Reply:
The `mode_auto` enter function would call `switch_state(ModeAuto::State::IDLE)`